pkg/metrics/service-monitor.go

```diff
@@ -33,6 +33,7 @@ func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Servic
 		return nil, err
 	}
 	if !exists {
+		log.Info("ServiceMonitor object could not be created as ServiceMonitor is not registered with the API")
```

Comment:
Since we're not exiting `main()` on an error on only logging, should we return this as an error and log it in `main.go`? I realize that's a breaking change for users who have found and are using this function, but maybe it's worth it to give callers more flexibility on how to handle it?

We could declare `ErrServiceMonitorNotPresent` (or something) so that users could check the error in their calling code if they wanted to have custom logic?

Reply:
This is not an error but a warning, just because they do not have the prom-operator installed in their cluster we should not be throwing an error here. The whole point of checking through [the list of APIs](https://github.com/operator-framework/operator-sdk/blob/3f675d697274fa88c9fd92c0bb11e4eeb016c499//pkg/k8sutil/k8sutil.go#L75:6) for the ServiceMonitor was to not produce many errors to the api server, so we shouldn't do one here either. We log it as informational, but I would not return error here.

Reply:
Lili and I chatted a bit offline. The functions called from the SDK-scaffolded `main.go` are kind of hybrid helper functions and libraries, with no clear delineation.

This function and leader election (and probably others) have checks that can log and disable the function's intended purpose within the function, making them more helper functions that libraries.

Typically, libraries don't log and they return errors when the thing you asked them to do can't be done. So we have this question about what to do with these helpers.

My two cents is that we should make them more like libraries for two primary reasons:
1. It makes them more flexible outside the context of a `main.go` function.
2. To make the case for upstreaming these into `controller-runtime` we need to make sure we conform to other controller-runtime style and design choices, which seems more library-centric than `main.go`-helper-centric.

@LiliC does this capture our convo accurately?

Reply:
Yep, that sounds about right, also opened the issue so we don't forget to do this, put the milestone as well. 

Will make the changes for this function.